src/ORM/FieldType/DBComposite.php

```diff
@@ -240,8 +240,9 @@ public function setField($field, $value, $markChanged = true)
     {
         $this->objCacheClear();
 
-        // Skip non-db fields
+        // Non-db fields get assigned as normal properties
         if (!$this->hasField($field)) {
+            $this->$field = $value;
```

Comment:
Can we at least use `return parent::setField($field, $value)` and minimise direct access to unknown variables.